extensions/data-plane-transfer/data-plane-transfer-sync/src/main/java/org/eclipse/dataspaceconnector/transfer/dataplane/sync/DataPlaneTransferSyncExtension.java

```diff
@@ -0,0 +1,153 @@
+        manifestGenerator.registerProviderGenerator(new HttpProviderProxyResourceGenerator());
+
+        var statusCheckerReg = context.getService(StatusCheckerRegistry.class);
+        statusCheckerReg.register(HttpProxySchema.TYPE, new HttpProviderProxyStatusChecker());
+
+        registerProvisioner(context);
+        registerDataFlowController(context);
+        context.getTypeManager().registerTypes(HttpProviderProxyProvisionedResource.class, HttpProviderProxyResourceDefinition.class);
+
+        // consumer part
+        registerTransformer(context);
+    }
+
+    /**
+     * Resolve public key from the Vault.
+     */
+    private static PublicKeyResolver createResolver(ServiceExtensionContext context) {
+        var vault = context.getService(Vault.class);
+        var publicKeyAlias = getMandatorySetting(context, PUBLIC_KEY_ALIAS);
+        return new SelfPublicKeyResolver(vault, publicKeyAlias);
+    }
+
+    /**
+     * Register {@link EndpointDataReferenceTransformer} allowing to use consumer data plane as proxy.
+     */
+    private void registerTransformer(ServiceExtensionContext context) {
+        var endpoint = getMandatorySetting(context, DATA_PLANE_PUBLIC_API_ENDPOINT);
+        var consumerProxyTransformer = new ProxyEndpointDataReferenceTransformer(tokenGenerationService, dataEncrypter, endpoint, context.getTypeManager());
+        context.registerService(EndpointDataReferenceTransformer.class, consumerProxyTransformer);
+    }
+
+    /**
+     * Register provider proxy provisioner serving data.
+     */
+    private void registerProvisioner(ServiceExtensionContext context) {
+        var provisionManager = context.getService(ProvisionManager.class);
+        var endpoint = getMandatorySetting(context, DATA_PLANE_PUBLIC_API_ENDPOINT);
+        var tokenValidity = context.getSetting(DATA_PLANE_PUBLIC_API_TOKEN_VALIDITY_SECONDS, DEFAULT_DATA_PLANE_PUBLIC_API_TOKEN_VALIDITY_SECONDS);
+        var provisioner = new HttpProviderProxyProvisioner(endpoint, dataAddressResolver, dataEncrypter, tokenGenerationService, tokenValidity, context.getTypeManager());
+        provisionManager.register(provisioner);
+    }
+
+    /**
+     * Register data flow controller in charge of sending to the consumer EDC the
+     * {@link org.eclipse.dataspaceconnector.spi.types.domain.edr.EndpointDataReference} which describes the endpoint serving the data.
+     * Note that default protocol for exchange is `ids-multipart`.
+     */
+    private void registerDataFlowController(ServiceExtensionContext context) {
+        var dataFlowManager = context.getService(DataFlowManager.class);
+        dataFlowManager.register(new HttpProviderProxyDataFlowController(context.getConnectorId(), dispatcherRegistry));
+    }
+
+    private static @NotNull String getMandatorySetting(ServiceExtensionContext context, String name) {
+        var setting = context.getSetting(name, null);
```

Comment:
`context.getConfig().getString(name)` already throws an exception if the config is null